pandas/_libs/tslibs/np_datetime.pyx

```diff
@@ -78,6 +78,13 @@ cdef NPY_DATETIMEUNIT get_datetime64_unit(object obj) noexcept nogil:
     return <NPY_DATETIMEUNIT>(<PyDatetimeScalarObject*>obj).obmeta.base
 
 
+cdef int get_datetime64_unit_num(object obj) noexcept nogil:
+    """
+    returns the number of units of the dtype for a numpy datetime64 object.
+    """
+    return (<PyDatetimeScalarObject*>obj).obmeta.num
```

Comment:
What does number of units mean exactly? And is there a public way to get this - I am wary of relying up internals llike this